tests/unit/s2n_security_policies_test.c

```diff
@@ -107,147 +107,29 @@ int main(int argc, char **argv)
-                    }
-
-                    struct s2n_cert_chain_and_key *default_cert;
-                    EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&default_cert, S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN, S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
-
-                    EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, default_cert));
-                    EXPECT_FALSE(config->is_rsa_cert_configured);
-
-                    struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT);
-                    struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
-                    EXPECT_NOT_NULL(client_conn);
-                    EXPECT_NOT_NULL(server_conn);
-
-                    EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, security_policy_selection[policy_index].version));
-                    EXPECT_SUCCESS(s2n_config_set_verification_ca_location(config, S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN, NULL));
-                    EXPECT_NOT_NULL(config->default_certs_by_type.certs[S2N_PKEY_TYPE_ECDSA]);
-                    EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
-                    EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
-
-                    client_conn->actual_protocol_version = S2N_TLS13;
-                    server_conn->actual_protocol_version = S2N_TLS13;
-                    client_conn->secure->cipher_suite = &tls_13_ciphers[i];
-                    server_conn->secure->cipher_suite = &tls_13_ciphers[i];
+        if (!has_tls_13_cipher) {
+            continue;
+        }
 
-                    const struct s2n_signature_scheme *chosen_scheme = NULL;
+        bool has_tls_13_sig_alg = false;
+        bool has_rsa_pss = false;
 
-                    /* If an ECDSA Certificate is configured, then we should always be able to pick a default Signature
-                     * Scheme (even if RSA PSS is not supported by the libcrypto) */
-                    EXPECT_SUCCESS(s2n_tls13_default_sig_scheme(client_conn, &chosen_scheme));
-                    EXPECT_SUCCESS(s2n_tls13_default_sig_scheme(server_conn, &chosen_scheme));
+        for (size_t i = 0; i < security_policy->signature_preferences->count; i++) {
+            int min = security_policy->signature_preferences->signature_schemes[i]->minimum_protocol_version;
+            int max = security_policy->signature_preferences->signature_schemes[i]->maximum_protocol_version;
+            s2n_signature_algorithm sig_alg = security_policy->signature_preferences->signature_schemes[i]->sig_alg;
 
-                    EXPECT_SUCCESS(s2n_connection_free(client_conn));
-                    EXPECT_SUCCESS(s2n_connection_free(server_conn));
-                    EXPECT_SUCCESS(s2n_config_free(config));
-                    EXPECT_SUCCESS(s2n_cert_chain_and_key_free(default_cert));
-                }
+            if (min == S2N_TLS13 || max >= S2N_TLS13) {
+                has_tls_13_sig_alg = true;
             }
```

Comment:
Is a signature algorithm considered TLS1.3-capable if `min < S2N_TLS13` and `max == S2N_UNKNOWN_PROTOCOL_VERSION`?

Reply:
No, because at the moment that means it just wasn't updated when we added TLS1.3 :) In reality just checking min should be enough. I think we were just trying to be future-proof here?

But I didn't modify this test at all either, I just changed the indent.

Reply:
I think you're right, and I think there are other issues with this test. However, I didn't actually modify this test, just indent it ;) I've reverted that indentation change so that it's untouched by this PR, and I'll address the actual issues in a separate PR.

Reply:
I'll open this PR after this is merged: https://github.com/aws/s2n-tls/commit/669daed32040ba4208a85dbf2d1eadc30c645274